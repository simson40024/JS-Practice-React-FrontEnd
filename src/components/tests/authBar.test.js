import AuthBar from "../header/authBar";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

const InitialState = {
  user: {
    value: {
      name: "",
      token: "",
    },
  },
};

const AuthState = {
  user: {
    value: {
      name: "John Smitt",
      token: "UserToken",
    },
  },
};

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const mock = jest.spyOn(hooks, "useAppSelector");
const mockDispatch = jest.spyOn(hooks, "useAppDispatch");

describe("AuthBar", () => {
  it("should create AuthBar with not authorized user", () => {
    mock.mockImplementation((selector) => selector(InitialState));
    mockDispatch.mockReturnValue(jest.fn());
    act(() => {
      createRoot(container).render(<AuthBar />);
    });
    expect(container).toMatchSnapshot();
  });

  it("should create AuthBar with authorized user", () => {
    mock.mockImplementation((selector) => selector(AuthState));
    mockDispatch.mockReturnValue(jest.fn());
    act(() => {
      createRoot(container).render(<AuthBar />);
    });
    expect(container).toMatchSnapshot();
  });
});
