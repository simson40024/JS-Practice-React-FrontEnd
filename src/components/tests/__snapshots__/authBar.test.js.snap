// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthBar should create AuthBar with authorized user 1`] = `
<div>
  <div
    class="authorization"
  >
    <div
      id="authorization"
      style="color: darkgreen;"
    >
      Authorized user: John Smitt
    </div>
    <button
      id="authorization-btn"
      style="display: initial;"
    >
      logOut
    </button>
  </div>
</div>
`;

exports[`AuthBar should create AuthBar with not authorized user 1`] = `
<div>
  <div
    class="authorization"
  >
    <div
      id="authorization"
      style="color: white;"
    >
      User not authorized
    </div>
    <button
      id="authorization-btn"
      style="display: none;"
    >
      logOut
    </button>
  </div>
</div>
`;
